### ISSUE TITLE
7.1.0 crashes app on SIGPIPE when it was previously set to ignored.

### ISSUE THREAD

**Issue by martinbonaciugome** [facebook/facebook-ios-sdk#1444]
- Xcode Version: `11.5`
- Installation Platform & Verison: `Cocoapods` version `no idea`

## Expected Results

not crash on signals

## Actual Results

crashes on SIGPIPE

## Steps to Reproduce

cause a SIGPIPE

## Code Samples & Details

this is your code below in FBSDKCrashHandler.m
```
+ (void)installSignalHandler
{
  signal(SIGBUS, FBSDKSignalHandler);
  signal(SIGFPE, FBSDKSignalHandler);
  signal(SIGILL, FBSDKSignalHandler);
  signal(SIGPIPE, FBSDKSignalHandler);
  signal(SIGSEGV, FBSDKSignalHandler);
  signal(SIGSYS, FBSDKSignalHandler);
}

void FBSDKSignalHandler(int sig)
{
  NSMutableArray<NSString *> *callStack = [[NSThread callStackSymbols] mutableCopy];
  if (callStack) {
    NSIndexSet *indexSet = [NSIndexSet indexSetWithIndexesInRange:NSMakeRange(0, 2)];
    if (callStack.count > 2 && [callStack objectsAtIndexes:indexSet]) {
      [callStack removeObjectsAtIndexes:indexSet];
    }
  }
  [FBSDKCrashHandler saveSignal:sig withCallStack:callStack];

  // reset to default handler
  signal(sig, SIG_DFL);
  // re-signal to default handler
  raise(sig);
}
```

The amount of bugs in just those few lines is amazing. Whole concept is dangerous, and breaks other crash handlers.
SERIOUSLY JUST DON'T!!!
HIRE DEVELOPERS WITH EXPERIENCE IN CRASH HANDLERS!!!

**Comment by trzecieu** [comment]
I think that title is a little misleading. The problem is general how crash reporting is implemented (as stated in the description).
And correctly noticed, it basically disables any custom mechanisms of crash handling for client applications or games. 

For the contrast - ObjectiveC exception handling is implemented in a correct - non-intrusive way, and it respects what handler was set before - also it executes it. https://github.com/facebook/facebook-ios-sdk/blob/master/FBSDKCoreKit/FBSDKCoreKit/Basics/Instrument/FBSDKCrashHandler.m#L146-L168

Going back to handling signals:
* it would be much more flexible to use sigaction instead of signals.
* if some concrete signal was marked already with `SIG_IGN` this should be respected. 
* forwarding signals to previous handlers should recognize `sa_sigaction` from `sa_handler` basing on `sa_flags | SA_SIGINFO`

**Please note**: I would highly suggest increasing priority on that issue, as FBSDK 7 is to be required in terms of iOS14 support and soon this issue will be in the center of attention.

Good reference can be find:
* https://www.gnu.org/software/libc/manual/html_node/Basic-Signal-Handling.html

**Comment by joesus** [comment]
Should be fixed by dfc1cbbeb38487e30588d6b6e33a4e86f8f70ec2
